safe/gis/generic_expressions.py

```diff
@@ -252,14 +253,21 @@ def beautify_date(inasafe_time, feature, parent):
     """
     _ = feature, parent  # NOQA
     try:
-        datetime_object = datetime.datetime.strptime(
-            inasafe_time, '%Y-%m-%dT%H:%M:%S.%f')
-    except ValueError:
+        gmt_offset_str = inasafe_time[-6:-4]
```

Comment:
This looks dangerous. Is there not a function in datetime module to get the offset?

Reply:
Agree with @Gustry , perhaps you can check this one https://stackoverflow.com/questions/12281975/convert-timestamps-with-offset-to-datetime-obj-using-strptime

Reply:
I did a lot of googling for this and not sure there is such a function in datetime library.

Reply:
Ok, strange from a famous python module.
So make it safer as possible and future proof ;-)

Can you move the logic to safe.utilities folder in a single function (linked to the comment from Ismail) and add some quick tests?

Reply:
thanks @ismailsunni @Gustry , ok so with `parse` method we could generally convert it into the datetime object. but my intention was for knowing how much the time difference is so we could convert it to localtime. we could figure it put later